File: hono-app/src/dead/unused.ts
/**
 * Dead routes — NOT_REACHABLE.
 *
 * This file defines Hono routes but is NEVER imported in index.ts.
 * All findings here should be NOT_REACHABLE.
 */
import { Hono } from 'hono';
import * as serialize from 'node-serialize';

const deadRoutes = new Hono();

// SECRET: Dead credential (NOT_REACHABLE — routes never mounted)
const DEAD_ADMIN_TOKEN = 'ghp_deadDeadDeadDeadDeadDeadDeadDeadDead';

deadRoutes.post('/dead-deserialize', async (c) => {
  /** CVE-2017-5941 (node-serialize) — NOT_REACHABLE: never mounted. */
  const body = await c.req.text();
  return c.json(serialize.unserialize(body));
});

deadRoutes.get('/dead-admin', (c) => {
  /** SECRET — NOT_REACHABLE: never mounted. */
  return c.json({ token: DEAD_ADMIN_TOKEN });
});

export { deadRoutes };

File: echo-app/dead/unused.go
// Dead handlers — NOT_REACHABLE.
//
// These handlers are defined but NEVER registered in main.go.
// The scanner should classify all findings here as NOT_REACHABLE.
package dead

import (
	"fmt"
	"net/http"
	"os/exec"

	"github.com/labstack/echo/v4"
	"golang.org/x/text/language"
)

// SECRET: Dead database password (NOT_REACHABLE — handler never registered)
const DeadDatabasePassword = "postgres://admin:SuperSecret123@db.internal:5432/prod"

// DeadTranslate — CVE-2022-32149 (x/text) — NOT_REACHABLE: never registered.
func DeadTranslate(c echo.Context) error {
	tag, _ := language.Parse(c.FormValue("lang"))
	return c.JSON(http.StatusOK, map[string]string{"lang": tag.String()})
}

// DeadCommand — CWE-78 (command injection) — NOT_REACHABLE: never registered.
func DeadCommand(c echo.Context) error {
	cmd := c.QueryParam("cmd")
	out, _ := exec.Command("sh", "-c", cmd).Output() // CWE NOT_REACHABLE
	return c.String(http.StatusOK, fmt.Sprintf("%s", out))
}
